components/backup-stream/src/router.rs

```diff
@@ -708,22 +708,25 @@ impl TempFileKey {
 
     /// The full name of the file owns the key.
     fn temp_file_name(&self) -> String {
+        let timestamp = (|| {
+            fail::fail_point!("temp_file_name_timestamp", |t| t
+                .and_then(|v| {
+                    // reduce the precision of timestamp
+                    v.parse::<u64>().ok().map(|u| TimeStamp::physical_now() / u)
+                })
+                .unwrap_or(0));
```

Comment:
It seems we can use [`map_or_else`](http://doc.rust-lang.org/1.72.0/std/result/enum.Result.html#method.map_or_else) here.